### PR TITLE
Fix option update

### LINKED ISSUE #610: Handle options changes
Only the Controls properly handles options changes.

We should have something like:
```javascript
model_events() {
  let key;
  let options = this.model.get('options');
  for (var i = 0; i < o.length; i++) {
    key = options[i];
    this.listenTo(this.model, 'change:' + key, () => {
      L.setOptions(this.obj, this.get_options());
    }, this);
  }
}
```

On the Layer base class and on the Map.

### PR DESCRIPTION
**Pull request by davidbrochart**
Fixes #610

### CODE REVIEW

**Comment by martinRenou** [comment]
Does it work correctly?

There is now some code duplication, `layers/Velocity.js`, `layers/Heatmap.js` and `layers/Path.js` seem to implement it already so we might want to remove it there.

**Comment by davidbrochart** [comment]
> Does it work correctly?

Yes I checked on the Map and some layers and it seems to work fine.

> There is now some code duplication, `layers/Velocity.js`, `layers/Heatmap.js` and `layers/Path.js` seem to implement it already so we might want to remove it there.

Good point.

File: js/src/layers/Path.js [hunk 1]
```diff
@@ -27,19 +27,6 @@ export class LeafletPathModel extends vectorlayer.LeafletVectorLayerModel {
...
     super.model_events();
-    var key;
-    var o = this.model.get('options');
-    for (var i = 0; i < o.length; i++) {
-      key = o[i];
-      this.listenTo(
-        this.model,
-        'change:' + key,
-        function() {
-          this.obj.setStyle(this.get_options());
```
Comment (martinRenou) [inline js/src/layers/Path.js:38]: Did you check that this was still working? It's not exactly the same code.

Comment (davidbrochart) [reply js/src/layers/Path.js:38]: Good point, it doesn't work. It works if I keep `Path.js` as it were, but the `LeafletLayerView` still calls `L.setOptions` on the `LeafletPathView`. It doesn't seem to hurt, but is it okay?

Comment (martinRenou) [reply js/src/layers/Path.js:38]: Let's keep it as it was then.. It is a bit annoying. The API of LeafletJS is too heterogeneous.

Comment (davidbrochart) [reply js/src/layers/Path.js:38]: Do you mean we should close this PR?

Comment (martinRenou) [reply js/src/layers/Path.js:38]: No I mean let's keep `Path.js` as it was with the `setStyle`, but keep your other changes.

Comment (davidbrochart) [reply js/src/layers/Path.js:38]: :+1:

File: js/src/layers/Path.js [hunk 2]
```diff
@@ -27,6 +27,19 @@ export class LeafletPathModel extends vectorlayer.LeafletVectorLayerModel {
...
+      this.listenTo(
+        this.model,
+        'change:' + key,
+        function() {
+          this.obj.setStyle(this.get_options());
+        },
+        this
+      );
+    }
     this.obj.setStyle(this.get_options());
```
Comment (martinRenou) [inline js/src/layers/Path.js]: This line is unexpected :open_mouth: It should not be needed

Comment (martinRenou) [reply js/src/layers/Path.js]: Let's remove it later

Comment (davidbrochart) [reply js/src/layers/Path.js]: :+1:

**Pull request merged**